benchmarks/python/test_huggingface_attn_fwd.py

```diff
@@ -103,3 +120,28 @@ def test_huggingface_attn_fwd_benchmark(
 
     if not disable_benchmarking:
         run_benchmark(benchmark, fd.execute, [attention_mask, inputs])
+
+
+@pytest.mark.parametrize("compile", [False, True], ids=["eager", "compile"])
+@pytest.mark.parametrize("size", generate_attn_inputs())
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_huggingface_attn_fwd_baseline_benchmark(
+    benchmark,
+    size: tuple,
+    dtype: torch.dtype,
+    compile: bool,
+):
+    clear_cuda_cache()
+
+    batch_size, seq_len, nh, n_embd = size
+    dropout_p = 0.0
```

Comment:
is there an influence on perf if dropout is larger than 0, e.g. 0.2? If has, we can use a more practical value and set to 0 if validation is turned on.

Reply:
If its `0`, the predication is trivial and may influence the perf.
```
   S9 = fd.define_scalar(1 - dropout_p, dtype=DataType.Double)
    T10 = fd.ops.lt(T8, S9)
```

Reply:
Right. Let me make that change. Thanks!

Reply:
This would although mean that we benchmark different things based on whether validation is turned on.

Reply:
can you check the influence on bandwidth? I am not sure how much it may have, maybe very small.

Reply:
What do you think about having another fusion definition with 0.0 dropout in validation and 0.2 otherwise. That would ensure we always benchmark with dropout (0.2) and it is consistent with torch baseline benchmarks.

Reply:
Dropout: 0.2
![Screenshot 2024-04-04 at 11 53 57 AM](https://github.com/NVIDIA/Fuser/assets/52657555/0adf9ff0-811b-4512-a6a9-1c958669ca50)

Dropout: 0.0
![Screenshot 2024-04-04 at 11 53 49 AM](https://github.com/NVIDIA/Fuser/assets/52657555/ee1ef950-c084-46b4-a93b-f237021774c4)

The difference for eager is significant.

Reply:
> What do you think about having another fusion definition with 0.0 dropout in validation and 0.2 otherwise. That would ensure we always benchmark with dropout (0.2) and it is consistent with torch baseline benchmarks.

sounds good to me.

Reply:
> Dropout: 0.2 ![Screenshot 2024-04-04 at 11 53 57 AM](https://private-user-images.githubusercontent.com/52657555/319736946-0adf9ff0-811b-4512-a6a9-1c958669ca50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyNTkyOTksIm5iZiI6MTcxMjI1ODk5OSwicGF0aCI6Ii81MjY1NzU1NS8zMTk3MzY5NDYtMGFkZjlmZjAtODExYi00NTEyLWE2YTktMWM5NTg2NjljYTUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDE5Mjk1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5YmRmMjUzNzZhYTc1ODhkMjY0YzUxYTkyZTNjMmZiZGJkNmEyZjk4Mzk0ZmFmOTYxNzc0ZmY1ZjBhZTQ2MzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g9kgsWCSZF_WksLoaZMgAuHcSCWYQmrdlfAW4yoGFfg)
> 
> Dropout: 0.0 ![Screenshot 2024-04-04 at 11 53 49 AM](https://private-user-images.githubusercontent.com/52657555/319736994-ee1ef950-c084-46b4-a93b-f237021774c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyNTkyOTksIm5iZiI6MTcxMjI1ODk5OSwicGF0aCI6Ii81MjY1NzU1NS8zMTk3MzY5OTQtZWUxZWY5NTAtYzA4NC00NmI0LWE5M2ItZjIzNzAyMTc3NGM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDE5Mjk1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMzM5OGMyMTllZjg3MWFjZjBmMmU0ZjI1MDM4YjczYzRkYTMyYzNjZTlhNWQ2MGNhMzcyOGU0ZGE2ZDViZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OWxJoj0beWfZvLpahf70eOveRZzpxRuo62FyprTdBXU)
> 
> The difference for eager is significant.

This seems difficult to explain. can you create an issue to investigate further when we have time?

Reply:
Eager mode is probably skipping the dropout call altogether when `dropout=0` but i would guess torch.compile is either not skipping dropout, or it's just io-bound so skipping dropout makes no difference.

Reply:
that's very smart!